### DISCUSSION TITLE
Notifications for Mimir alerts

### DISCUSSION THREAD

**Discussion by sbanka** [grafana/mimir#1975] [General]
I have configured Mimir using the Getting started guide. Everything seems to work except notifications.
Here are the steps I took on empty running Mimir setup:

- added contact point with discord type
- tested new contact point - message appeared in discord channel
- in notification policies, I changed root policy to use new contact point
- added Mimir alert
- when alert triggered NO notification was sent to discord

Am I doing something wrong?
Does Grafana built-in alertmanager support Mimir alerts at all?
I checked pure Grafana alerts and these seem to work ok and notifications are sent.

I also tried to install Mimir alertmanager but I got precisely the same issues as described here: https://github.com/grafana/mimir/discussions/1931

All in all, I cannot make any notifications to be sent from Mimir based alerts.

my setup:
- Mimir 2.1.0
- Grafana 9.0.0-63885pre

**Comment by pracucci** [comment]
A couple of preliminary questions:
1. Could you share your Mimir configuration (redacting any sensitive information)?
2. The Mimir alertmanager exposes the Alertmanager UI at `/alertmanager`: if you open it with a web browser, does it show the firing alert?

**Reply by sbanka** [reply]
Here it is:

1. Mimir config: [mimir-config.txt](https://github.com/grafana/mimir/files/8815563/mimir-config.txt)
2. Mimir alertmanager returns only this: "no org id"

If I should see alertmanager UI I definitely don't see it...

**Reply by kotchaosu** [reply]
Similar situation in my setup described in (helm chart deployment): https://github.com/grafana/mimir/discussions/1931

Config file: [mimir.txt](https://github.com/grafana/mimir/files/8834427/mimir.txt)
Response from `/alertmanager`: **"the Alertmanager is not configured"** and in the nginx I see:

```
127.0.0.1 - - [03/Jun/2022:17:12:21 +0000]  404 "GET /alertmanager HTTP/1.1" 35 "-" [TRIMMED CONTENT...]
```

**Reply by pracucci** [reply]
> Mimir alertmanager returns only this: "no org id"

Mimir alertmanager supports multi-tenancy like any other Mimir component. When multi-tenancy is enabled, any request to Mimir is expected to contain the `X-Scope-OrgID: <tenant ID>` header in the HTTP request (see [doc here](https://grafana.com/docs/mimir/v2.1.x/operators-guide/securing/authentication-and-authorization/)). We provide a simple [tenant injector tool](https://grafana.com/docs/mimir/v2.0.x/operators-guide/tools/tenant-injector/) for testing purposes, even if typically you want to have your authentication gateway in front of Mimir, in order to authenticate requests and inject `X-Scope-OrgID` header based on the authenticated tenant.

Alternative, if you don't need multi-tenancy, you can consider disabling it. If you disable multi-tenancy, then `X-Scope-OrgID` header is not required (and will be ignored if provided):

```
# When set to true, incoming HTTP requests must specify tenant ID in HTTP
# X-Scope-OrgId header. When set to false, tenant ID from -auth.no-auth-tenant
# is used instead.
# CLI flag: -auth.multitenancy-enabled
[multitenancy_enabled: <boolean> | default = true]

# (advanced) Tenant ID to use when multitenancy is disabled.
# CLI flag: -auth.no-auth-tenant
[no_auth_tenant: <string> | default = "anonymous"]
```

All Mimir configuration parameters are documented [here](https://grafana.com/docs/mimir/v2.0.x/operators-guide/configuring/reference-configuration-parameters/).

> If I should see alertmanager UI I definitely don't see it...

You're not seeing it because you're getting the "no org id" error. The reason is explained ☝️ .
